sql/src/main/java/com/rilldata/calcite/CalciteToolbox.java

```diff
@@ -131,6 +165,61 @@ public SqlValidator getValidator(PlannerImpl planner) throws NoSuchFieldExceptio
     return (SqlValidator) validatorField.get(planner);
   }
 
+  static class Statement
+  {
+    public List<Statement> dependencies;
+    SqlNode node;
+    String ddl;
+    String name;
+    boolean changed;
+
+    public String getName()
+    {
+      return name;
+    }
+
+    public String getType()
+    {
+      if (node instanceof SqlCreateTable) {
+        return "TABLE";
+      } else if (node instanceof SqlCreateView)
+        return "VIEW";
+      return null;
+    }
+  }
+
+  public static List<MigrationStep> inferMigrations(String newSql, String schema, SqlDialect sqlDialect)
+      throws SqlParseException, JsonProcessingException
+  {
+    SqlNode node = parseStmts(newSql);
+
+    Map<String, Statement> existing = toStatementsMap(schema, sqlDialect);
+    createGraph(existing);
```

Comment:
whats the purpose of this ? can we add step by step algorithm in a comment

Reply:
it calls DependencyFinder to figure out dependencies between database entities

Reply:
Added.